commit/tokenprice/outcome.go

```diff
@@ -0,0 +1,151 @@
+package tokenprice
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-ccip/shared"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+)
+
+// getConsensusObservation Combine the list of observations into a single consensus observation
+func (p *processor) getConsensusObservation(
+	aos []shared.AttributedObservation[Observation],
+) (ConsensusObservation, error) {
+	aggObs := aggregateObservations(aos)
+
+	if len(aggObs.FDestChain) < p.bigF {
+		return ConsensusObservation{},
+			fmt.Errorf("not enough observations for fDestChain")
+	}
+
+	fDestChain := shared.Median(aggObs.FDestChain, func(a, b int) bool {
+		return a < b
+	})
+
+	twoFPlus1 := 2*fDestChain + 1
```

Comment:
nit, but this is also something that I see a lot in different places, do you think it's worth exporting to enum or function? 

```go
twoFPlus1 = function (int) int
```

Don't we have a similar concept somewhere in the libocr? Could that be reused or sth?

Reply:
I like the idea, let me try it out. 👍